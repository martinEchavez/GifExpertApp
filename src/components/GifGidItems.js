import React from 'react'
import PropTypes from 'prop-types'

const GifGidItems = ({ title, url }) => {

    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

export default GifGidItems

GifGidItems.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}