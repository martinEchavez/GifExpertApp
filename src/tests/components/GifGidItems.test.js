import React from 'react'
import { shallow } from 'enzyme'
import GifGidItems from '../../components/GifGidItems'

describe('Pruebas en <GifGidItems />', () => {

    const title = 'Titulo de prueba'
    const url = 'https//localhosto/prueba'
    // Se obtienen las caracteristicas del componente con los parámetros que recibe
    const wraper = shallow( <GifGidItems title={title} url={url}/> )

    test('Debe mostrar el componente correctamente', () => {
        expect(wraper).toMatchSnapshot();// Copia del cuerpo del componente, en caso que ocurra algún cambio este va a notificar
    })

    test('El title debe ser un párrafo ', () => {
        const p = wraper.find('p')// Se obtiene el parrafo y se verifica que coincida con la variable title
        expect(p.text().trim()).toBe(title)
    })

    test('src y alt de la imagen igual a las props ', () => {
        const img = wraper.find('img')
        //console.log(img.props().src)// Se verifica que las propiedades de la etiqueta img conincidan
        expect(img.props().src).toBe(url)
        expect(img.props().alt).toBe(title)
    })

    test('Debe terner la clase animate__fadeIn', () => {
        const div = wraper.find('div')
        const className = div.props().className // Se ontiene la clase animate__fadeIn
        expect(className.includes('animate__fadeIn')).toBe(true) // Se verifica que la clase animate__fadeIn esté en el componente
    })
})