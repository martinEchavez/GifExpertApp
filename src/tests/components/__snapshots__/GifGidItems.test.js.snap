// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGidItems /> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo de prueba"
    src="https//localhosto/prueba"
  />
  <p>
    Titulo de prueba
  </p>
</div>
`;
